### PR TITLE
refactor: adjust decision def and req search domain classes

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

Same as https://github.com/camunda/camunda/pull/24750
applied to:
- Decision definition
- Decision requirements

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to https://github.com/camunda/camunda/issues/24911

### CODE REVIEW

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionDefinitionFieldSortingTransformer.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import static io.camunda.webapps.schema.descriptors.operate.index.DecisionIndex.KEY;
+
+public class DecisionDefinitionFieldSortingTransformer implements FieldSortingTransformer {
+
+  @Override
+  public String apply(final String domainField) {
+    return switch (domainField) {
+      case "decisionDefinitionKey" -> KEY;
+      case "decisionDefinitionId" -> DECISION_ID;
+      default -> domainField;
```
Comment (stoerti) [inline search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionDefinitionFieldSortingTransformer.java]: 🔧  list all possible fields and throw exception (as in `DecisionInstanceFieldSortingTransformer`)

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionRequirementsFieldSortingTransformer.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import static io.camunda.webapps.schema.descriptors.operate.index.DecisionRequirementsIndex.KEY;
+
+public class DecisionRequirementsFieldSortingTransformer implements FieldSortingTransformer {
+
+  @Override
+  public String apply(final String domainField) {
+    return switch (domainField) {
+      case "decisionRequirementsKey" -> KEY;
+      default -> domainField;
```
Comment (stoerti) [inline search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionRequirementsFieldSortingTransformer.java]: 🔧  list all possible fields and throw exception (as in `DecisionInstanceFieldSortingTransformer`)

File: search/search-domain/src/main/java/io/camunda/search/entities/DecisionDefinitionEntity.java [hunk 3]
```diff
@@ -12,8 +12,8 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public record DecisionDefinitionEntity(
     String tenantId,
```
Comment (stoerti) [inline search/search-domain/src/main/java/io/camunda/search/entities/DecisionDefinitionEntity.java:14]: 💭  tenantId as first property makes it look like this is the PK. Is it maybe useful to change it that the PK is the first argument?

File: search/search-domain/src/main/java/io/camunda/search/entities/DecisionRequirementsEntity.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public record DecisionRequirementsEntity(
     String tenantId,
```
Comment (stoerti) [inline search/search-domain/src/main/java/io/camunda/search/entities/DecisionRequirementsEntity.java:14]: 💭  tenantId as first property makes it look like this is the PK. Is it maybe useful to change it that the PK is the first argument?

**Review by stoerti — APPROVED** [review]
Looks good to me 👍

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionDefinitionFieldSortingTransformer.java (hunk 1 above)
Comment (houssain-barouni) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionDefinitionFieldSortingTransformer.java]: yes I just did that before the review :) 
so, I have force pushed the changes

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionRequirementsFieldSortingTransformer.java (hunk 2 above)
Comment (houssain-barouni) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/sort/DecisionRequirementsFieldSortingTransformer.java]: yes I just did that before the review :) 
so, I have force pushed the changes

**Pull request merged**